hardware_interface/CMakeLists.txt

```diff
@@ -36,6 +37,7 @@ ament_target_dependencies(
   hardware_interface
   control_msgs
   pluginlib
+  rclcpp_lifecycle
```

Comment:
I am not sure if we really want that dependency on the hardware layer.

Reply:
Is this possible, since we are inheriting from `LifecycleNodeInterface`?